### ISSUE TITLE
rerun-triggers are not handed over to the cluster

### ISSUE THREAD

**Issue by timtroendle** [snakemake/snakemake#1679]
**Snakemake version**
7.8.0

**Describe the bug**
I am running on a cluster (LSF) and I am removing "software-env" from the new rerun-triggers (because it behaves strangely). Sometimes, a scheduled job runs more than one rule which never happened before. I believe this is because my rerun-triggers parameterisation is not applied within the job.

**Version**
7.8.0

**Logs**

<details><summary>Cluster log</summary>
<p>

Job AAA was submitted from host XXXX by user YYYY in cluster ZZZ at Fri May 27 14:47:44 2022

....

Started at Fri May 27 14:47:53 2022
Terminated at Fri May 27 14:53:22 2022
Results reported at Fri May 27 14:53:22 2022

....

[snakemake call without "rerun-triggers"]

....

[Fri May 27 14:47:56 2022]
Job 1: XXX
Reason: Software stack has changed since last execution

....

[Fri May 27 14:52:57 2022]
Finished job 1.
1 of 2 steps (50%) done
Select jobs to execute...

[Fri May 27 14:52:57 2022]
Job 0: YYYY
Reason: Input files updated by another job: some-output

....

[Fri May 27 14:53:21 2022]
Finished job 0.
2 of 2 steps (100%) done

</p>
</details>

**Minimal example**

**Additional context**

**Comment by chrarnold** [comment]
In which version these rerun triggers were introduced? I don't find a proper description of them in the Docs, did I oversee it? If specifying the rerun triggers is not translating to cluster mode properly, 7.8 is unusable for me because of too many unnecessary reruns that don't result in any change in the result files

**Comment by timtroendle** [comment]
Hi @chrarnold , they were introduced in 7.8 and they are described here:
https://snakemake.readthedocs.io/en/stable/executing/cli.html

I think they are great in general, but this bug means I cannot use them in my cluster setup right now.

**Comment by chrarnold** [comment]
Thanks for the pointer, quite hidden there, but a good and useful addition indeed, I agree! Nevertheless, as this is a breaking change between Snakemake 7.8 and all previous versions, I think it should be highlighted, linked, and explained more than it currently is. Most users wont find this change or will b confused, not knowing they can restore the old behavior with this new argument

**Comment by johanneskoester** [comment]
Hi folks. Thanks a lot, indeed they should be propagated, working on that now. Also working on the bug that causes false triggers by the software stack.

**Comment by johanneskoester** [comment]
Yes, that was a bug in how needrun determination and conda env assignment was ordered.
